orchestrator/src/main/java/eu/arrowhead/core/orchestrator/OrchestratorController.java

```diff
@@ -256,6 +344,41 @@ private void checkOrchestratorFormRequestDTO(final OrchestrationFormRequestDTO r
 		}
 	}
 	
+	//-------------------------------------------------------------------------------------------------	
+	private void checkOrchestratorStoreFlexibleRequestDTOList(final List<OrchestratorStoreFlexibleRequestDTO> dtoList, final String origin) {
+		logger.debug("checkOrchestratorStoreFlexibleRequestDTOList started...");
+		
+		if (dtoList == null) {
+			throw new BadPayloadException("Request list" + NULL_PARAMETER_ERROR_MESSAGE, HttpStatus.SC_BAD_REQUEST, origin);
+		}
+		
+		for (final OrchestratorStoreFlexibleRequestDTO dto : dtoList) {
+			if (dto == null) {
+				throw new BadPayloadException("Request list contains null element", HttpStatus.SC_BAD_REQUEST, origin);
+			}
+			
+			if (dto.getConsumerSystem() == null) {
+				throw new BadPayloadException("Request list contains an element without consumer system describer", HttpStatus.SC_BAD_REQUEST, origin);
+			} else {
+				if(Utilities.isEmpty(dto.getConsumerSystem().getSystemName()) && (dto.getConsumerSystem().getMetadata() == null || dto.getConsumerSystem().getMetadata().isEmpty())) {
```

Comment:
I suggest a new method in Utilities: isEmpty on maps which handles null value as well